### PR TITLE
Node 0.12, iojs support

### PR DESCRIPTION
**Pull request by kuba-kubula**
- `+` a small bonus of a test with context separation
- `+` renaming pitboss into pitboss-ng (next gen.) and prepared to publish it into NPM

### CODE REVIEW

File: src/pitboss-ng.coffee [hunk 1]
```diff
@@ -80,22 +81,24 @@ exports.Runner = class Runner extends EventEmitter
     error = @currentError || "Process Failed"
     @emit 'failed', error
     @callback(error) if @callback
-    @notifyCompleted()
+    @notifyCompleted() if @callback
 
   timeout: =>
-    @currentError = "Timedout"
+    @currentError ?= "Timedout"
```
Comment (abtris) [inline src/pitboss-ng.coffee]: Can be better `Timed out` or `Timeout`

**Comment by abtris** [comment]
I don't see anything blocking. Free to merge and publish.

File: src/pitboss-ng.coffee (hunk 1 above)
Comment (Almad) [inline src/pitboss-ng.coffee]: I wouldn't overwrite the codebase there, though

**Comment by Almad** [comment]
Looks good™

**Comment by kuba-kubula** [comment]
@abtris @Almad thanks for review, will publish once tests pass in merged master as `pitboss-ng`. Will ping ops with follow-ups for upgrading in core//other apps

**Pull request merged**